ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/web3j/Web3JClient.java

```diff
@@ -36,17 +38,23 @@ public abstract class Web3JClient {
 
   private final EventLogger eventLog;
   private final TimeProvider timeProvider;
+  private final ExecutionClientEventsChannel executionClientEventsPublisher;
   private Web3jService web3jService;
   private Web3j eth1Web3j;
 
   // Default to the provider having a previous failure at startup so we log when it is first
   // available but uses a very old value to make sure we log if the first request fails
   private final AtomicLong lastError = new AtomicLong(STARTUP_LAST_ERROR_TIME);
   private boolean initialized = false;
+  private final AtomicBoolean isAvailable = new AtomicBoolean(false);
```

Comment:
Shouldn't need this new boolean as we can just use the `lastError` field same as we check it in `handleSuccess` and `handleError`. Should be able to just call the publisher at the same point we're calling `eventLog.executionClientRecovered` or `logExecutionClientError`